include/getusername.h

```diff
@@ -0,0 +1,55 @@
+#ifndef GETUSERNAME_H
+#define GETUSERNAME_H
+#ifdef HAVE_UNISTD_H
+ #include <unistd.h>
+#endif
+#ifdef _WIN32
+ #include <io.h>
+#else
+ #include <pwd.h>
+#endif
+#include <pthread_port.h>
+
+#define GETUSERNAME(user_p) GETUSERNAME_BEGIN(user_p);GETUSERNAME_END;
+
+#define GETUSERNAME_BEGIN(user_p) {\
```

Comment:
Why are these split into two macros?

Reply:
it is used in two parts in one file.. need tro check where. so its a constructor;
then the code can safely do stuff with teh static username
and a cleanup.